cargo-miri/bin.rs

```diff
@@ -398,22 +461,25 @@ fn in_cargo_miri() {
         return;
     }
 
+    let (target_filters, cargo_args, miri_args) =
+        parse_cargo_miri_args(std::env::args().skip(skip));
+
     // Now run the command.
     for target in list_targets() {
-        let mut args = std::env::args().skip(skip);
         let kind = target
             .kind
             .get(0)
             .expect("badly formatted cargo metadata: target::kind is an empty array");
+        if !target_filters.matches(kind, &target.name) {
+            continue;
+        }
         // Now we run `cargo check $FLAGS $ARGS`, giving the user the
         // change to add additional arguments. `FLAGS` is set to identify
         // this target.  The user gets to control what gets actually passed to Miri.
         let mut cmd = cargo();
         cmd.arg("check");
         match (subcommand, kind.as_str()) {
             (MiriCommand::Run, "bin") => {
-                // FIXME: we just run all the binaries here.
```

Comment:
The first line here is still true -- `cargo miri run` runs all binaries when there are several, where `cargo run` picks `default-run` or errors of that key is not specified in `Cargo.toml`. So please don't fully remove this FIXME.